app/presenters/report_presenter.rb

```diff
@@ -34,26 +34,18 @@ def due_defects_by_priority(priority:)
   end
 
   def overdue_defects_by_priority(priority:)
-    defects = defects_by_priority(priority: priority)
-    defects_completed_late = defects.where('target_completion_date < actual_completion_date')
-    defects_still_open_and_overdue = defects.open.where('target_completion_date < ?', Date.current)
-
-    (defects_completed_late + defects_still_open_and_overdue).uniq
+    (
+      defects_completed_late(priority: priority) +
+      defects_still_open_and_overdue(priority: priority) +
+      completed_defects_with_no_completion_date(priority: priority)
+    ).uniq
   end
 
   def defects_completed_on_time(priority:)
     completed_defects(priority: priority).select do |completed_defect|
-      completed_defect_activities = completed_defect.activities.where(key: 'defect.update')
-      activities_on_time = completed_defect_activities.select do |activity|
-        activity.created_at.to_date <= completed_defect.target_completion_date
-      end
-
-      # TODO: Query parameter JSON at database level rather than in Ruby
-      true if activities_on_time.detect do |update_activity|
-        update_activity.parameters &&
-        update_activity.parameters[:changes] &&
-        update_activity.parameters[:changes][:status]&.last == 'completed'
-      end
+      next if completed_defect.actual_completion_date.nil?
+
+      completed_defect.actual_completion_date <= completed_defect.target_completion_date
```

Comment:
Feels like this change could have been a separate commit from the other. I was confused why the overdue method had changed from the commit message title.

Reply:
Have move the overdue changes into the commit that fixes overdue behaviour.